### DISCUSSION TITLE
Vignette Inverse or Radial Gradient - Adjusting exposure for darkened corners

### DISCUSSION THREAD

**Discussion by adamvanspronsen** [ImageMagick/ImageMagick#7238] [Digital Image Processing]
Hi there,
Anyone know how to adjust images that have darkened corners?  I would like to autotone many images that have some lens shading in the corners.  In Adobe Lightroom, I can apply a radial gradient for exposure to adjust this - but I need a script to run on a computer that will not have Lightroom and was hoping something like ImageMagick could work.
Any ideas or solutions for this?
Thanks.

**Comment by fmw42** [comment]
Imagemagick can create and apply radial gradients.  Please post an example image.

**Comment by adamvanspronsen** [comment]
![ImageExample-LensShading](https://github.com/ImageMagick/ImageMagick/assets/85695982/d8307904-fb31-4d58-a272-baa63d68d3c0)

Here is an example Image.  As you can see the corners are quite dark.  I would like to auto-tone the images and adjust this dark corners from lens shading. 

Any help you can provide would be greatly appreciated.

**Reply by fmw42** [reply]
How would you do that in Photoshop?  I don't know of any anti-vignette tool?  In Imagemagick you would need to create an appropriate radial gradient and then use that to apply some process to the image. I do not know how to figure out what radial gradient is needed automatically.

**Reply by fmw42** [reply]
You could also create your radial gradient using either:  -vignette or radial-gradient: with an appropriate define. See https://imagemagick.org/Usage/canvas/#radial-gradient and PSEUDO-IMAGE FORMATS section at https://imagemagick.org/script/defines.php and https://imagemagick.org/script/command-line-options.php#vignette and https://imagemagick.org/Usage/transform/#vignette

**Reply by adamvanspronsen** [reply]
Thanks for this information, I will try this out today.

**Comment by snibgo** [comment]
If you don't know the vignetting applicable to your lens, the easiest way is to photograph a plain white card. Then divide any photograph taken with that lens by the image of the white card.

**Reply by fmw42** [reply]
See https://imagemagick.org/Usage/photos/#vignettation for that process

**Reply by adamvanspronsen** [reply]
Thanks for this information, I will try this out today.

**Comment by adamvanspronsen** [comment]
_> How would you do that in Photoshop? I don't know of any anti-vignette tool? In Imagemagick you would need to create an appropriate radial gradient and then use that to apply some process to the image. I do not know how to figure out what radial gradient is needed automatically._

In Adobe Lightroom, there is a radial gradient option, then adjust exposure for that gradient.

**Reply by fmw42** [reply]
But you still need to know the correct radial gradient for the image vignette.
